flow/flow.go

```diff
@@ -309,28 +310,43 @@ func (p *Packet) TransportFlow() (gopacket.Flow, error) {
 	return layer.TransportFlow(), nil
 }
 
+//Hash flow with custom func
+func Hash(f gopacket.Flow, hasher hash.Hash) {
+	if strings.Compare(string(f.Src().Raw()), string(f.Dst().Raw())) > 0 {
+		hasher.Write(f.Src().Raw())
+		hasher.Write(f.Dst().Raw())
+	} else {
+		hasher.Write(f.Dst().Raw())
+		hasher.Write(f.Src().Raw())
+	}
+
+	value64 := make([]byte, 8)
+	binary.BigEndian.PutUint64(value64, uint64(f.EndpointType()))
+	hasher.Write(value64)
+}
+
 // Key returns the unique flow key
 // The unique key is calculated based on parentUUID, network, transport and applicable layers
 func (p *Packet) Key(parentUUID string, opts Opts) string {
-	var uuid uint64
-
+	hasher := xxHash64.New(0)
 	// uses L2 is requested or if there is no network layer
 	if opts.LayerKeyMode == L2KeyMode || p.NetworkLayer() == nil {
 		if layer := p.LinkLayer(); layer != nil {
-			uuid ^= layer.LinkFlow().FastHash()
+			Hash(layer.LinkFlow(), hasher)
+			hasher.Write([]byte(strconv.FormatInt(linkID(p), 10)))
```

Comment:
use binary package to convert int64 to bytes
     b := make([]byte, 8)
    binary.BigEndian.PutUint64(b, linkID(p))

Reply:
done